### ISSUE TITLE
Parties API Returning Attorneys and Party Types for Other Cases

### ISSUE THREAD

**Issue by a-drumm** [freelawproject/courtlistener#3858]
Examples:

https://www.courtlistener.com/api/rest/v3/parties/?docket=4508405
https://www.courtlistener.com/api/rest/v3/parties/?docket=6283353
https://www.courtlistener.com/api/rest/v3/parties/?docket=6280169
https://www.courtlistener.com/api/rest/v3/parties/?docket=4263779

Several are the City of Chicago, and the other attorney/party_types are from other cases where the City of Chicago is also a party. Some though have attorney/party_types for cases that do not appear to even be the same party. 

Only one of these I was able to find the page for on courtlisterner.com (sans API) and none of those extra attorney/party_types show up.

I don't know the system well enough to have any idea what is going on, but it does not seem intentional.

**Comment by a-drumm** [comment]
Right sorry. I couldn't find some of the docket pages through searching the case name + district court. Shows up if I go right to the page. Thank you for that direction.

Moving on to the action issue: 

I apologize, I was being lazy and I came at this the wrong way before.

Specific example:

I ran a query for parties in Rodriguez v. Chicago: https://www.courtlistener.com/api/rest/v3/parties/?docket=4263779

The 13th result is “Chad Smith” and gives the resource uri https://www.courtlistener.com/api/rest/v3/parties/68837/

That party id has party types/associated attorneys for 7 different cases (type of party and court location are provided for indication that these are almost certainly not the same person):

Chad Smith, a defendant, case in ND Alabama: https://www.courtlistener.com/docket/6185451/parties/lockhart-v-franklin/ 

Chad Smith, plaintiff, case in D. Colorado: https://www.courtlistener.com/docket/4195548/parties/abeyta-v-cfi-steel-lp/

Chad Smith, defendant, case in E.D. Michigan: https://www.courtlistener.com/docket/4832214/parties/daniel-v-granholm/

Chad Smith, plaintiff, case in ED Louisiana: https://www.courtlistener.com/docket/6301012/parties/brewer-v-bp-plc/

Chad Smith, plaintiff, case in CD California: https://www.courtlistener.com/docket/4145474/parties/jeffory-weaver-v-county-of-orange/

Chad Smith, plaintiff, case in SD Texas: https://www.courtlistener.com/docket/4397937/parties/allen-v-city-of-texas-city/

Chad Smith, defendant, ND Illinois: https://www.courtlistener.com/docket/4263779/rodriguez-v-city-of-chicago/


It appears like the same party ID is being given in cases where not only is it not the same case, but in this instance not even the same person. 

Sometimes it is the same party, just not the same case. e.g. party id [150980](https://www.courtlistener.com/api/rest/v3/parties/150980/) (City of Chicago) connects to 219 distinct cases. 

If you need more please let me know. Again, apologize for the initial lack of specificity.

**Comment by mlissner** [comment]
Thanks for those details. Yeah, it looks like we're using the same "Chad Smith" (https://www.courtlistener.com/api/rest/v3/parties/68837/) in multiple entirely unrelated dockets in a way that is wrong. It's pretty obvious that the Chad Smith in these two cases is not the same person:

 - https://www.courtlistener.com/docket/4263779/parties/rodriguez-v-city-of-chicago/ (ilnd)
 - https://www.courtlistener.com/docket/4397937/parties/allen-v-city-of-texas-city/ (txsd)

This isn't how the system is supposed to work and I don't know of a bug that would have caused this. I do notice though that these cases are all pretty old. It's possible this problem arose some time ago during an import.

I'm going to go purchase the party info and see if it gets automatically fixed.

**Comment by mlissner** [comment]
Hm, I bought and RECAP'ed both of these cases and it didn't fix the issue. This is worth more investigation to see if it's happening now (I doubt it), or if it's "just" the case that older cases don't get updated when new data comes in.

Another thing that surprises me is that the Chad Smith in the Chicago case has a "Start number" in PACER, but that never seems to go into RECAP:

![image](https://github.com/freelawproject/courtlistener/assets/236970/569bd216-0406-40e3-9ee3-9b95ae713fe3)

**Comment by a-drumm** [comment]
I believe that's the Officer Star #, like a badge ID. 

And this doesn't affect my ability to do anything, I just have to dump all the case info I don't want. Did confuse the hell out of me at first though.

**Comment by mlissner** [comment]
Yeah, I guess the fix would be to split up these kinds of parties, but I actually thought that was how it already worked, which is why I think it's only (mainly?) affecting older cases?

**Comment by a-drumm** [comment]
I found this after matching data from City on cases terminated w/ settlements from 2019-2022. No newer cases had anything like this, so the only affecting older cases thing seems reasonable.

**Comment by mlissner** [comment]
Sorry, when you say "older" can you clarify what you mean? I think I mean before 2016 or so? Are you saying you see this in your 2019-2022 cases?

**Comment by a-drumm** [comment]
I have cases that terminated in 2019-2022, by older I'm referring to ones that were filed about 2016 or earlier.

**Comment by mlissner** [comment]
OK. So I think the first task here, in that case, is just to make sure new additions are working properly. If so, we can investigate how big the problem is and think through how to solve it. Thanks @a-drumm.

**Comment by tactipus** [comment]
hi guys! Do you still require help on this issue?

**Comment by mlissner** [comment]
@tactipus, I think this is a pretty complicated one, so I think it's best if we handle it in house until you get more comfortable with the code. I'm going to assign it to one of our devs for mid-late January, but if you get to it before then — great! 

You might find the voluneer backlog to be interesting though: https://github.com/orgs/freelawproject/projects/31/views/1

**Comment by tactipus** [comment]
@mlissner thanks! I'll check it out!

**Comment by tactipus** [comment]
@mlissner i'll have to find a way to star or bookmark that backlog. I am positive you mentioned it before.
